libcontainer/system/linux.go

```diff
@@ -130,6 +130,33 @@ func RunningInUserNS() bool {
 	return true
 }
 
+// GetParentNSeuid returns the euid within the parent user namespace
+func GetParentNSeuid() int {
```

Comment:
I think @brauner had a patch like this for Ubuntu? This code should also check `/proc/self/setgroups` and ensure that it's set to `deny`.

Reply:
As commented above, I think we need to find an alternative check to support nested userns